libs/agno/agno/agent/agent.py

```diff
@@ -1182,6 +1191,11 @@ async def _arun(
         else:
             # Get the model response
             model_response = await self.model.aresponse(messages=run_messages.messages)
+            # Format tool calls if they exist
+            if model_response.tool_calls:
+                model_response.format_tool_calls()
+                self.run_response.formatted_tool_calls = model_response.formatted_tool_calls
```

Comment:
Do you need to have `formatted_tool_calls` on the model_response as well ?

Reply:
that's where I'm storing the extracted tool call info in this fashion `get_current_stock_price(symbol=TSLA)`
extracted using (tool_name and args) using the function in `response.py`
```py
def format_tool_calls(self) -> None:
        """Format tool calls for display in a readable format."""
        self.formatted_tool_calls = []
        for tool_call in self.tool_calls:
            if "tool_name" in tool_call and "tool_args" in tool_call:
                tool_name = tool_call["tool_name"]
                args_str = ", ".join(f"{k}={v}" for k, v in tool_call["tool_args"].items())
                self.formatted_tool_calls.append(f"{tool_name}({args_str})")
```

Reply:
Instead of storing it, how about that function just returns the formatted tool calls and you set it directly on the run_response? So no intermediate storage. The less we store on the message the better, because it will be serialized, etc. And takes up memory, etc